maven-plugin-report-plugin/src/main/java/org/apache/maven/plugin/plugin/report/AbstractPluginReportRenderer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.plugin.plugin.report;
+
+import java.util.Locale;
+
+import org.apache.maven.doxia.sink.Sink;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.reporting.AbstractMavenReportRenderer;
+import org.codehaus.plexus.i18n.I18N;
+
+/** Base class for all reports generated by the plugin report plugin. Encapsulates i18n functionality as well as functionality to directly
+ * emit XHTML markup */
+public abstract class AbstractPluginReportRenderer extends AbstractMavenReportRenderer {
+
+    private static final String KEY_PREFIX = "report.plugin.";
+
+    private static final String RESOURCE_BASENAME = "plugin-report";
+
+    private final I18N i18n;
+
+    protected final Locale locale;
+
+    protected final MavenProject project;
+
+    protected AbstractPluginReportRenderer(Sink sink, Locale locale, I18N i18n, MavenProject project) {
+        super(sink);
+        this.i18n = i18n;
+        this.locale = locale;
+        this.project = project;
+    }
+
+    /** @param key The key.
+     * @return The translated string. */
+    protected String getI18nString(String key) {
+        return i18n.getString(RESOURCE_BASENAME, locale, KEY_PREFIX + key);
+    }
+}
```

Comment:
It should follow the pattern as in here: https://github.com/apache/maven-project-info-reports-plugin/blob/ec0918223c43e3135aa067c9fcdc6c0a2427f24a/src/main/java/org/apache/maven/report/projectinfo/AbstractProjectInfoRenderer.java, maybe even with `getI18Nsection()`. This stencil is used throughout all reporting plugins now, at least in the `doxia-2.0.0` branch.

Reply:
Done.